Sources/CrowdNotifierBaseSDK/CryptoUtils/CryptoUtilsBase.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  File.swift
+//  
+//
+//  Created by Matthias Felix on 15.04.21.
+//
+
+import Foundation
+import HKDF
+
+public final class CryptoUtilsBase {
+
+    private static let hkdfDomainKey = "CrowdNotifier_v3"
+
+    public static func getNoncesAndNotificationKey(infoBytes: Bytes) -> (nonce1: Bytes, nonce2: Bytes, notificationKey: Bytes)? {
+        // Length: 32 bytes each for nonce1, nonce2 & notification_key
```

Comment:
Should there be a test that the arguments are of correct length? Or are all callers to this method guaranteed to have the correct format?

Reply:
I'm not sure I understand which arguments you're talking about. The only argument `infoBytes` can be of arbitrary length, right?

Reply:
I mean, in practice they'll probabloy have some minimum length, but indeed, I don't think we can validate the input length in any meaning full way. I think it is fine to trust the callers to pass in something useful :D.

Reply:
Speaking of which: `infoBytes` -> `qrCodePayloadBytes` (or whatever we used in the other implementations)?

Reply:
Yes, I'll change that to the name we agreed on, as soon as I find out which one it is ;-)

Reply:
I thought of the nonce1, nonce2 and the notification key.

Reply:
@ineiti those 3 values are _outputs_ (so no test required then right?)

Reply:
Ok, I have changed all occurences of `infoBytes` to `qrCodePayload`.

Reply:
> @ineiti those 3 values are _outputs_ (so no test required then right?)

Of course - sorry for that...